crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -155,119 +198,272 @@ struct SensitivePattern {
         freq[byte as usize] += 1;
     }
 
-    let mut entropy = 0.0;
+    let mut shannon_entropy = 0.0;
+    let mut letter_count = 0;
+    let mut uppercase_count = 0;
+    let mut lowercase_count = 0;
+    let mut digit_count = 0;
+    let mut symbol_count = 0;
+    let mut case_switches = 0;
+    let mut previous_char_was_upper = false;
+
     for count in freq.iter() {
         if *count > 0 {
             let p = *count as f64 / len as f64;
-            entropy -= p * p.log2();
+            shannon_entropy -= p * p.log2();
         }
     }
 
-    entropy
+    // Letter classification and case switching
+    for (i, c) in data.chars().enumerate() {
+        if c.is_ascii_alphabetic() {
+            letter_count += 1;
+            if c.is_uppercase() {
+                uppercase_count += 1;
+                if i > 0 && !previous_char_was_upper {
+                    case_switches += 1;
+                }
+                previous_char_was_upper = true;
+            } else {
+                lowercase_count += 1;
+                if i > 0 && previous_char_was_upper {
+                    case_switches += 1;
+                }
+                previous_char_was_upper = false;
+            }
+        } else if c.is_ascii_digit() {
+            digit_count += 1;
+        } else if !c.is_whitespace() {
+            symbol_count += 1;
+        }
+    }
+
+    // Adjust entropy: case switches and symbol boosts
+    let case_entropy_boost = if uppercase_count > 0 && lowercase_count > 0 {
+        (case_switches as f64 / letter_count as f64) * 2.0
+    } else {
+        0.0
+    };
+
+    let symbol_entropy_boost = if symbol_count > 0 {
+        symbol_count as f64 / len as f64
+    } else {
+        0.0
+    };
+
+    let digit_entropy_boost = if digit_count > 0 {
+        digit_count as f64 / len as f64
+    } else {
+        0.0
+    };
+
+    let adjusted_entropy = shannon_entropy
+        + (case_entropy_boost * 2.5)
+        + (symbol_entropy_boost * 1.5)
+        + digit_entropy_boost;
+
+    // Apply exponential scaling to avoid excessive boosting for long, structured tokens
+    apply_exponential_entropy_scaling(adjusted_entropy, len, base_threshold, scaling_factor)
+}
+
+fn apply_exponential_entropy_scaling(
+    entropy: f64,
+    token_length: usize,
+    base_threshold: f64,
+    scaling_factor: f64,
+) -> f64 {
+    // We will apply a logarithmic dampening to prevent excessive scaling for long tokens
+    let scaling_adjustment = (token_length as f64 / scaling_factor).ln();
+    base_threshold + entropy * scaling_adjustment
+}
+
+fn split_into_tokens(value: &str) -> Vec<&str> {
+    let delimiters = [' ', '\t', '\n', '.', ',', ';', ':', '/', '-', '_', '@'];
+    let mut tokens = vec![value];
+
+    for &delimiter in delimiters.iter() {
+        tokens = tokens
+            .into_iter()
+            .flat_map(|token| token.split(delimiter).filter(|&s| !s.is_empty()))
+            .collect();
+    }
+
+    tokens
```

Comment:
Have you tried [str::split](https://doc.rust-lang.org/std/primitive.str.html#method.split)?

Reply:
Works. Went for that instead, thanks for catching.